packages/ERTP/src/displayInfo.js

```diff
@@ -0,0 +1,38 @@
+import { assert, details, q } from '@agoric/assert';
+
+/**
+ * Assert all values from `part` appear in `whole`.
+ *
+ * @param {string[]} whole
+ * @param {string[]} part
+ */
+export const assertSubset = (whole, part) => {
+  part.forEach(key => {
+    assert.typeof(key, 'string');
+    assert(
+      whole.includes(key),
+      details`key ${q(key)} was not one of the expected keys ${q(whole)}`,
+    );
+  });
+};
+
+// Assert that the keys of `record` are all in `allowedKeys`. If a key
+// of `record` is not in `allowedKeys`, throw an error. If a key in
+// `allowedKeys` is not a key of record, we do not throw an error.
+export const assertKeysAllowed = (allowedKeys, record) => {
+  const keys = Object.getOwnPropertyNames(record);
+  assertSubset(allowedKeys, keys);
+  // assert that there are no symbol properties.
+  assert(
+    Object.getOwnPropertySymbols(record).length === 0,
+    details`no symbol properties allowed`,
+  );
+};
+
+export const assertDisplayInfo = allegedDisplayInfo => {
```

Comment:
There are a number of ways that `allegedDisplayInfo` may misbehave that this code does not protect against. The simplest and safest defense would be to first make a `pureCopy` using https://github.com/Agoric/agoric-sdk/blob/76e0723915a32b6b76ebea55e4df29d67f8afea8/packages/marshal/src/marshal.js#L30-L32